packages/core/src/providers/starknet.tsx

```diff
@@ -165,18 +173,31 @@ function useStarknetManager({
     dispatch({ type: 'set_account', account: undefined })
     dispatch({
       type: 'set_provider',
-      provider: userDefaultProvider ? userDefaultProvider : defaultProvider,
+      provider: userDefaultProvider ? userDefaultProvider : customDefaultProvider,
     })
     dispatch({ type: 'set_connector', connector: undefined })
     if (autoConnect) {
       localStorage.removeItem('lastUsedConnector')
     }
     if (!state.connector) return
-    state.connector.removeEventListener(handleAccountChanged)
-    state.connector.disconnect().catch((err) => {
-      console.error(err)
-      dispatch({ type: 'set_error', error: new ConnectorNotFoundError() })
-    })
+    state.connector.disconnect().then(
+      () => {
+        dispatch({ type: 'set_account', account: undefined })
+        dispatch({
+          type: 'set_provider',
+          provider: userDefaultProvider ? userDefaultProvider : customDefaultProvider,
+        })
+        dispatch({ type: 'set_connector', connector: undefined })
+        state.connector?.removeEventListener(handleAccountChanged)
+        if (autoConnect) {
+          localStorage.removeItem('lastUsedConnector')
+        }
```

Comment:
Why do you repeat these calls after `disconnect` has been resolved? Is it really needed?

Reply:
I don't remember but indeed they don't seem useful I removed them